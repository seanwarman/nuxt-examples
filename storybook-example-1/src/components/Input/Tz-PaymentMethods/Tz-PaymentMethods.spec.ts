/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import TzPaymentMethods from './Tz-PaymentMethods.vue';
import { expect, describe, beforeEach, it } from 'vitest';

let wrapper: VueWrapper;

describe('TzPaymentMethods', () => {
  beforeEach(() => {
    wrapper = mount(TzPaymentMethods, {
      props: {
        name: 'x',
        label: 'y',
        options: [
          {
            value: 'a',
            label: 'A'
          },
          {
            value: 'b',
            label: 'B'
          }
        ]
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('gives a unique id to each radio input', () => {
    expect(wrapper.find('input#x_0').exists()).toBe(true);
    expect(wrapper.find('input#x_1').exists()).toBe(true);
  });

  it("links the label 'for' to each radio input 'id'", () => {
    expect(wrapper.find('label[for="x_0"]').exists()).toBe(true);
    expect(wrapper.find('label[for="x_1"]').exists()).toBe(true);
  });

  it('sets the values to each radio input', () => {
    expect(wrapper.find('input#x_0').attributes().value).toEqual('a');
    expect(wrapper.find('input#x_1').attributes().value).toEqual('b');
  });

  it('sets the name on each radio input', () => {
    expect(wrapper.find('input#x_0').attributes().name).toEqual('x');
    expect(wrapper.find('input#x_1').attributes().name).toEqual('x');
  });
});
