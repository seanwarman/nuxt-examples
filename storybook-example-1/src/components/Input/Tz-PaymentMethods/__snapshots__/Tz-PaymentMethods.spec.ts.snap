// Vitest Snapshot v1

exports[`TzPaymentMethods > renders correctly 1`] = `
<div
  data-v-637e0bdc=""
>
  <div
    class="text-xs font-black uppercase leading-10 tracking-wider text-dark-fade"
    data-v-637e0bdc=""
  >
    y
  </div>
  <div
    class="flex flex-col gap-5 border p-5 xs:gap-0 xs:border-none xs:p-0"
    data-v-637e0bdc=""
  >
    
    <div
      class="flex cursor-pointer flex-row items-center xs:h-[74px] xs:border-b"
      data-v-637e0bdc=""
    >
      <input
        class="peer mr-4 inline-block h-[18px] w-[18px] cursor-pointer"
        data-v-637e0bdc=""
        id="x_0"
        name="x"
        type="radio"
        value="a"
      />
      <label
        class="flex flex-1 cursor-pointer flex-row items-center align-middle text-black peer-checked:font-extrabold peer-checked:text-gray-800"
        data-v-637e0bdc=""
        for="x_0"
      >
        <span
          class="flex-1"
          data-v-637e0bdc=""
        >
          A
        </span>
        <span
          class="hidden flex-row gap-2 xs:flex"
          data-v-637e0bdc=""
        >
          
          
        </span>
      </label>
    </div>
    <div
      class="flex cursor-pointer flex-row items-center xs:h-[74px] xs:border-b"
      data-v-637e0bdc=""
    >
      <input
        class="peer mr-4 inline-block h-[18px] w-[18px] cursor-pointer"
        data-v-637e0bdc=""
        id="x_1"
        name="x"
        type="radio"
        value="b"
      />
      <label
        class="flex flex-1 cursor-pointer flex-row items-center align-middle text-black peer-checked:font-extrabold peer-checked:text-gray-800"
        data-v-637e0bdc=""
        for="x_1"
      >
        <span
          class="flex-1"
          data-v-637e0bdc=""
        >
          B
        </span>
        <span
          class="hidden flex-row gap-2 xs:flex"
          data-v-637e0bdc=""
        >
          
          
        </span>
      </label>
    </div>
    
  </div>
  <!--v-if-->
</div>
`;
